src/goto-programs/show_goto_functions_json.h

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************\
+
+Module: Goto Program
+
+Author: Thomas Kiley
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_PROGRAMS_SHOW_GOTO_FUNCTIONS_JSON_H
+#define CPROVER_GOTO_PROGRAMS_SHOW_GOTO_FUNCTIONS_JSON_H
+
+class goto_functionst;
+class namespacet;
+
+class show_goto_functions_jsont
+{
+public:
+  explicit show_goto_functions_jsont(const namespacet &ns);
+
+  void show_goto_functions(const goto_functionst &goto_functions);
```

Comment:
Can we add two variants for flexibility:
- one that returns the ``json_objectt`` instead of writing it to ``std::cout`` directly
- one that takes a ``std::ostream &`` and prints the output there